### PR TITLE
Disallow ability to create a conda env with `:` in the prefix

### PR DESCRIPTION
**Pull request by beeankha**
Resolves: https://github.com/conda/conda/issues/12749

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: conda/cli/main_create.py [hunk 1]
```diff
@@ -21,6 +22,12 @@
 
 @notices
 def execute(args, parser):
+    if not on_win:
+        if ":" in context.target_prefix:
+            raise CondaValueError(
+                "Cannot create a conda environment with a ':' in the prefix. Aborting."
+            )
+
```
Comment (dholth) [inline conda/cli/main_create.py:30]:
```suggestion
        if os.pathsep in context.target_prefix:
            raise CondaValueError(
                f"Cannot create a conda environment with '{os.pathsep}' in the prefix. Aborting."
            )

```

**Review by kenodegard — CHANGES_REQUESTED** [review]
this check belongs in the helper function `conda.cli.install.check_prefix`, this is where we already check for things like reserved environment names (`base`) and warn users if there's a space in the prefix:

https://github.com/conda/conda/blob/db8f1c3580bf597458bebc787498b05ae9a5f0c7/conda/cli/install.py#L52-L73

**Comment by beeankha** [comment]
pre-commit.ci autofix

File: conda/cli/install.py [hunk 2]
```diff
@@ -50,6 +50,10 @@
 
 
 def check_prefix(prefix, json=False):
+    if os.pathsep in context.target_prefix:
```
Comment (kenodegard) [inline conda/cli/install.py:53]: In this case the prefix is passed in:

```suggestion
    if os.pathsep in prefix:
```

File: tests/conda_env/test_create.py [hunk 3]
```diff
@@ -250,3 +253,16 @@ def test_create_update_remote_env_file(
...
+def test_fail_to_create_env_in_dir_with_colon(tmp_path: Path):
+    # Add a directory with a colon
+    colon_dir = tmp_path / "fake:dir"
+    colon_dir.mkdir()
+
+    with pytest.raises(
+        CondaValueError,
+        match="Cannot create a conda environment with ':' in the prefix.",
+    ):
+        run_command(Commands.CREATE, "colon_env", f"--prefix={colon_dir}/tester")
```
Comment (kenodegard) [inline tests/conda_env/test_create.py:268]: `run_command` is deprecated, please use `conda_cli` instead:

```suggestion
@pytest.mark.skipif(on_win, reason="Test is invalid on Windows")
def test_fail_to_create_env_in_dir_with_colon(tmp_path: Path, conda_cli: CondaCLIFixture):
    # Add a directory with a colon
    colon_dir = tmp_path / "fake:dir"
    colon_dir.mkdir()

    with pytest.raises(
        CondaValueError,
        match="Cannot create a conda environment with ':' in the prefix.",
    ):
        conda_cli("create", f"--prefix={colon_dir}/tester")
```

**Review by kenodegard — CHANGES_REQUESTED** [review]

File: conda/cli/install.py (hunk 2 above)
Comment (dholth) [reply conda/cli/install.py:53]: As long as prefix includes the whole installation path.

Comment (kenodegard) [reply conda/cli/install.py:53]: `check_prefix` is used in two places, in one case it passes in `context.target_prefix`, in the other it effectively passes it in:

https://github.com/conda/conda/blob/5c814f97ac02af80477eaf89229a7c2b66814ca0/conda/cli/install.py#L141-L143

https://github.com/conda/conda/blob/5c814f97ac02af80477eaf89229a7c2b66814ca0/conda_env/cli/main_create.py#L118-L122

**Comment by beeankha** [comment]
pre-commit.ci autofix

**Review by kenodegard — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Pull request merged**